linalg/hypre.hpp

```diff
@@ -234,22 +237,25 @@ class HypreParMatrix : public Operator
    /// An empty matrix to be used as a reference to an existing matrix
    HypreParMatrix();
 
-   /// Converts hypre's format to HypreParMatrix
-   /** If @a owner is false, ownership of @a a is not transferred */
-   explicit HypreParMatrix(hypre_ParCSRMatrix *a, bool owner = true)
+   void WrapHypreParCSRMatrix(hypre_ParCSRMatrix *a, bool owner = true)
    {
-      Init();
       A = a;
       if (!owner) { ParCSROwner = 0; }
       height = GetNumRows();
       width = GetNumCols();
    }
 
-   /// Creates block-diagonal square parallel matrix.
-   /** Diagonal is given by @a diag which must be in CSR format (finalized). The
-       new HypreParMatrix does not take ownership of any of the input arrays.
-       See @ref hypre_partitioning_descr "here" for a description of the row
-       partitioning array @a row_starts.
+   /// Converts hypre's format to HypreParMatrix
+   /** If @a owner is false, ownership of @a a is not transferred */
+   explicit HypreParMatrix(hypre_ParCSRMatrix *a, bool owner = true)
+   {
+      Init();
+      WrapHypreParCSRMatrix(a, owner);
+   }
+
+   /** Creates block-diagonal square parallel matrix. Diagonal is given by diag
```

Comment:
Some lines in the original comments here seem to be lost. If that is a mistake, please restore.